### ISSUE TITLE
CSize and Int literal direct comparison fails on X86

### ISSUE THREAD

**Issue by LeeTibbert** [scala-native/scala-native#455]
This is a 32 bit X86 issue, destined for backlog.

This may well be an instance of issue #445. That
issue was created after I did this debugging. IIUC,
after a fix to #445, the .toLong below is not wrong, but
is not the "most reduced, minimal code" form. The
CSize and Int literal may be compared directly.
In that case, the fix would be to delete the .toInt
and let the magic happen behind the scenes.

Without my local changes, the  32 bit
unit-tests/src/main/scala/scala/scalanative/native/CStringSuite.scala test "toCString" test
fails at: 
```
assert(strlen(cstrTo) == 5)
```
I believe that the (32 bit fixed) toCString() call which created
cstrTo is innocent. assert (strlen(c"12345") == 5) also
fails &  test validates that the bytes of the returned CString
are as expected.

The fix seems to be to add explicit .toLong
 as successfully used in
 nativelib/src/main/scala/scala/scalanative/native/package.scala
 
```   method fromCString():
         val len   = string.strlen(cstr).toInt
```
Note: I have quoted the code as it exists now but. Issue #454 
identifies a 64-bit bug suggests changing the .toInt to .toLong.

  So the assert in CStringSuite.scala becomes:
```
    assert(strlen(cstrTo).toLong == 5L) // L in 5L is critical
```
With this (double) fix, the test passes.

Lee

**Comment by LeeTibbert** [comment]
A placeholder to keep up with the gratifyingly fast pace of development.  Not intended
to be a developer time sink.

I tested the 32 bit path again after commit #514 "Add support for cooperative equality" fixed #445.
This issue remains, albeit in a slightly different form.

Works:
```
 assert(strlen(cstrTo).toInt == 5) // Works
```
Fails:
```
 assert(strlen(cstrTo) == 5) // gitHub standard code: FAILS
 assert(strlen(cstrTo).toLong == 5L) // FAILS
```

**Comment by densh** [comment]
I've created a big summary issue https://github.com/scala-native/scala-native/issues/727 to accumulate all reports on x86.
